src/protocol/sort/bit_permutations.rs

```diff
@@ -2,19 +2,20 @@ use crate::{
     error::BoxError,
     field::Field,
     helpers::mesh::{Gateway, Mesh},
-    protocol::{context::ProtocolContext, IPAProtocolStep, RecordId, SortStep},
+    protocol::{context::ProtocolContext, IPAProtocolStep, RecordId},
     secret_sharing::Replicated,
 };
-
+use embed_doc_image::embed_doc_image;
 use futures::future::try_join_all;
 
+use super::SortStep;
 #[derive(Debug)]
 pub struct BitPermutations<'a, F> {
     input: &'a [Replicated<F>],
 }
 
+/// Create an object to generate bit permutations for a given bit column of query. This is GENBITPERM(Algorithm 3) from the paper
```

Comment:
Future readers of your code will not know what "the paper" refers to. And are you sure it's still "Algorithm 3" is the new one?

Reply:
I will link to the paper. The new paper implements malicious whereas we are implementing semi honest for now. Once we get to malicous, will review the comments to ensure they are updated